### ISSUE TITLE
Ubuntu Bionic x86 builds fail

### ISSUE THREAD

**Issue by dnsmichi** [Icinga/icinga2#7350]
## Failure

https://git.icinga.com/packaging/deb-icinga2/-/jobs/14020

## Background

Something influences the compiler on Bionic x86 between git diff f9f86e3f2 1fc660dcd33993d054ee2f10ea4857b89c6f07f4.

@Al2Klimov we've discussed this, for some reason the changes with the reload handling have an influence here resulting in a gcc coredump. Please check whether some parsing/code needs to be fixed to workaround the gcc issue, and consult @lazyfrosch here.

**Comment by dnsmichi** [comment]
Maybe it is also related to concurrent compiles with make, https://github.com/opencv/opencv/issues/7636

**Comment by dnsmichi** [comment]
@Al2Klimov was bisecting all day long, thanks for that 👍 

```
9c923687746294293473c25f2821253a9f18b469 is the first bad commit
commit 9c923687746294293473c25f2821253a9f18b469
Author: Michael Friedrich <michael.friedrich@icinga.com>
Date:   Tue Jun 18 14:58:19 2019 +0200

    SSL Context: Explicitly load ECC ciphers on el7

    Otherwise curl/nss as client won't be able to use the
    new default cipher list.

    fixes #7247

:040000 040000 9e75a34f152ef58c8f486cbd0be4f47d645b5381 e31aea526645ebf99a1d04ff099c20e21ecaa9ce M    lib
```

Seems to be the case that `SSL_CTX_set_ecdh_auto` was deprecated and removed in OpenSSL 1.1.0.

https://github.com/openssl/openssl/issues/1437

So while our fix works well on RHEL7 fixing their bug with not loading the ECC thingy, it somehow influences Bionic builds on x86 where the macro is a no-op but somehow produces gcc coredumps.

All the patches for `SSL_CTX_set_ecdh_auto` implement an SSL version guard around it.

https://github.com/lighttpd/lighttpd1.4/commit/76b9b1fa46e82d6bd31b71e69ef0893098f5f6d5
